src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -118,7 +120,10 @@ public static PackageSpec GetPackageSpec(IEnumerable<IMSBuildItem> items)
 
             PackageSpec result = null;
 
-            var specItem = items.SingleOrDefault(item =>
+            // There should only be one ProjectSpec per project in the item set, 
+            // but if multiple do appear take only the first one in an effort
+            // to handle this gracefully.
+            var specItem = items.FirstOrDefault(item =>
```

Comment:
With some refactoring a logger could be passed around in these methods. I left that out for now.